internal/pkg/pkg.go

```diff
@@ -159,29 +154,63 @@ func (p *Pkg) RelativePathTo(ancestorPkg *Pkg) (string, error) {
 
 // SubPackages returns sub packages of a pkg.
 // A sub directory that has a Kptfile is considered a sub package.
-func (p *Pkg) SubPackages() (subPkgs []*Pkg, err error) {
+func (p *Pkg) SubPackages() ([]*Pkg, error) {
 	pkgPath := string(p.UniquePath)
 
-	files, err := ioutil.ReadDir(pkgPath)
-	if err != nil {
-		return subPkgs, fmt.Errorf("failed to read sub dirs for %s %w", p, err)
-	}
-	for _, f := range files {
-		if f.IsDir() {
-			dirPath := path.Join(pkgPath, f.Name())
-			isKptPkg, err := kptfileutil.HasKptfile(dirPath)
-			if err != nil {
-				return subPkgs, fmt.Errorf("failed to check kptfile in subpackage %w", err)
+	packagePaths := make(map[string]bool)
+	if err := filepath.Walk(pkgPath, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return fmt.Errorf("failed to read package %s: %w", p, err)
+		}
+
+		// Ignore the root folder
+		if path == pkgPath {
+			return nil
+		}
+
+		// Ignore anything inside the .git folder
+		rel := strings.TrimPrefix(path, pkgPath)
+		if copyutil.IsDotGitFolder(rel) {
+			return nil
+		}
+
+		// Any paths that are inside subpackages can be ignored since
+		// we only want the subpackages directly underneath the package.
+		for dir := range packagePaths {
```

Comment:
Can't you skip walking subpackages using filepath.SkipDir?

You should avoid doing path computations in favor of using just `info`